### ISSUE TITLE
DL3040 is triggered when cleaning the cache using 'rm' instead of 'dnf clean all'

### ISSUE THREAD

**Issue by soenkeliebau** [hadolint/hadolint#796]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
DL3040 should not throw a warning if the cache is cleaned by rm'ing the /var/cache/yum folder instead of a dnf clean all.

'rm -rf /var/cache/yum' actually _might_ clean more than a dnf clean, as it will also clean cached data for disabled and orphaned repos. But this statement is based on research of about 3 minutes and should be confirmed by people with more knowledge than me.

### Actual behavior
Hadolint warns about a missing clean all for the Dockerfile snippet given below.

### Steps to reproduce the behavior

Create dockerfile with the below dnf snippet and run Hadolint against it.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

This is run as a Github action with the [ReviewDog Hadolint Action](https://github.com/reviewdog/action-hadolint) 

Hadolint version: [2.8.0](https://github.com/reviewdog/action-hadolint/blob/2f36e7ddb63420bac655b14287cb6d6b82dd84e2/action.yml#L48)

```bash
uses reviewdog/action-hadolint@2f36e7ddb63420bac655b14287cb6d6b82dd84e2 
```

**Dockerfile** (if relevant)

```Dockerfile
...
# Update image
RUN dnf update --disablerepo=* --enablerepo=ubi-8-baseos --enablerepo=ubi-8-baseos -y \
  && rm -rf /var/cache/yum \
  && dnf install --disablerepo=* --enablerepo=ubi-8-baseos shadow-utils -y \
  && rm -rf /var/cache/yum
...
```

**Additional environment details (OS, stack version, etc.)**

**Comment by lorenzo** [comment]
@soenkeliebau would you like to contribute this feature? I can offer some guidance

**Comment by soenkeliebau** [comment]
I had a quick look at the code back when I opened this issue and have to say that I understood nothing of what is happening here :)
I am happy to give it a try of course, but I'm afraid the time committment I can promise is quite limited..

**Comment by lorenzo** [comment]
@soenkeliebau you just need to add and `||` (or) to this line https://github.com/hadolint/hadolint/blob/master/src/Hadolint/Rule/DL3040.hs#L29 to check if the rm command is present like we do in this line https://github.com/hadolint/hadolint/blob/6c4632387a3485a9b7e1bf46c58da697178e02e2/src/Hadolint/Rule/DL3009.hs#L89
